Development/src/dataProvider.js

```diff
@@ -10,14 +10,23 @@ import {
 } from 'react-admin';
 import get from 'lodash/get';
 import set from 'lodash/set';
-import { set as setJSON } from 'json-ptr';
+import setJSON from 'json-ptr';
 import assign from 'lodash/assign';
 import diff from 'deep-diff';
 import Cookies from 'universal-cookie';
 
 let API_URL = '';
 
-let LINK_HEADER = '';
+let LINK_HEADERS = {
```

Comment:
My feeling has always been that really the link headers ought not to be cached in the `dataProvider` but be passed out with the `dataProvider` response and passed back in with the next `dataProvider` request (rather than just `FIRST`, `NEXT`, etc.)

But if this solves a bug right now... please just explain the bug in more detail! :-)

Reply:
Explained in 1820511
The bug occurs when using a reference field in a list view.